gateway/src/project.rs

```diff
@@ -515,6 +515,10 @@ impl ProjectCreating {
                     ],
                     "Env": [
                         "RUST_LOG=debug",
+                        // This should be set to the same version we pin Rust to in our Containerfile.
+                        // If we don't set this, users' dependencies with a `rust-toolchain.toml`
+                        // override will compile with incompatible versions of rust.
+                        "RUSTUP_TOOLCHAIN=1.65"
```

Comment:
Is there a way to get this version from the currently installed toolchain? Ie get it from the environment to keep it in sync with the Containerfile?

Reply:
We can use [this crate](https://crates.io/crates/rustc_version_runtime) to get the version gateway is compiled with and pass it to the deployer container config. I'm not a 100% if this goes against the intentions of this crate, so alternatively we could set the `RUSTUP_TOOLCHAIN` env var from the `Makefile`, extract it with `std::env::var(RUSTUP_TOOLCHAIN)` and pass it to the config. What do you think is best here?

Reply:
I like the `Makefile` idea. We can go a step further and pass the value from `Makefile` in as a build arg to the `Containerfile`. In the `Containerfile` the build arg can then set both the image of rust being used as well as a `RUST_TOOLCHAIN` environment variable. Then we don't have to worry about setting it in gateway.